health_check/contrib/rabbitmq/backends.py

```diff
@@ -0,0 +1,51 @@
+import logging
+from kombu import Connection
+from amqp.exceptions import AccessRefused
+
+from django.conf import settings
+
+from health_check.backends import BaseHealthCheckBackend
+from health_check.exceptions import ServiceUnavailable
+
+logger = logging.getLogger(__name__)
+
+
+class RabbitMQHealthCheck(BaseHealthCheckBackend):
+    """
+    Healthchecker for RabbitMQ.
+    """
+
+    def check_status(self):
+        """
+        Checks if RabbitMQ service is up by opening and closing
+        a broker channel.
+        """
+        logger.info("Checking for a broker_url on django settings...")
+
+        broker_url = getattr(settings, "BROKER_URL", None)
+
+        try:
```

Comment:
There is too much happening in the try block. Please be more narrow, where you expect the exception to occour.